### PR TITLE
Support per-group flags files in partest

### PR DESCRIPTION
**Pull request by phaller**
In a directory-based test with file A_1.scala, scalac flags that should apply only to A_1.scala can be put in the file A_1.flags. Local flags are appended to global flags.

This capability is important for testing the inliner, for example.

Review by @paulp

### CODE REVIEW

**Comment by phaller** [comment]
PLS REBUILD ALL

File: src/partest/scala/tools/partest/nest/CompileManager.scala [hunk 1]
```diff
@@ -94,7 +95,17 @@ class DirectCompiler(val fileManager: FileManager) extends SimpleCompiler {
     val logFile = basename(log.getName)
     val flagsFileName = "%s.flags" format (logFile.substring(0, logFile.lastIndexOf("-")))
     val argString = (io.File(log).parent / flagsFileName) ifFile (x => updatePluginPath(x.slurp())) getOrElse ""
-    val allOpts = fileManager.SCALAC_OPTS.toList ::: argString.split(' ').toList.filter(_.length > 0)
+
+    // slurp local flags (e.g., "A_1.flags")
+    val fstFile = SFile(files(0))
+    def isInGroup(num: Int) = fstFile.stripExtension endsWith ("_" + num)
+    val inGroup = (1 to 9) flatMap (group => if (isInGroup(group)) List(group) else List())
```
Comment (jsuereth) [inline src/partest/scala/tools/partest/nest/CompileManager.scala:102]: Would it make sense to detect groups rather than hard code 1-9?

Comment (paulp) [inline src/partest/scala/tools/partest/nest/CompileManager.scala:102]: I'm not sure it matters, unless you are planning on a lot of 10+ group tests.  I've not seen more than 3, I don't think.

**Comment by paulp** [comment]
This looks fine to me (assuming it accomplishes the intended goal - I haven't tested it or anything.)

File: src/partest/scala/tools/partest/nest/CompileManager.scala (hunk 1 above)
Comment (jsuereth) [inline src/partest/scala/tools/partest/nest/CompileManager.scala:102]: Ok.  When we see the "WHY ISN'T THIS WORKING" email, I'll cc you.

**Pull request merged**